bundle/phases/deploy.go

```diff
@@ -143,12 +143,21 @@ func deployCore(ctx context.Context, b *bundle.Bundle) {
 	}
 }
 
+// uploadLibraries uploads libraries to the workspace.
+// It also cleans up the artifacts directory and transforms wheel tasks.
+// It is called by only "bundle deploy".
+// Not having TransformWheelTask in plan phase might lead to jobs using these feature not
+// being reported in plan as being updated. This is acceptable since TransformWheelTask is
+// an experimental feature.
```

Comment:
> This is acceptable since TransformWheelTask is an experimental feature.

I don't think that makes it acceptable. This PR is a gradual improvement so we scan still land it and discuss follow up afterwards. The end goal is for plan to have the same view as deploy.

Reply:
Also libraries.Upload mutates the config so it needs to be split as well.

Reply:
Extracted it into a separate mutator here, so all config mutating operations are included in plan
https://github.com/databricks/cli/pull/3442/commits/22a6ea6016769b8f83f945b99fa1af8f72f9481d